californium-core/src/main/java/org/eclipse/californium/core/network/Matcher.java

```diff
@@ -353,18 +367,24 @@ public Exchange receiveResponse(Response response) {
 				// See issue #275
 				return null;
 			}
-
-			// we have received a Response matching the token of an ongoing Exchange's Request
-			// according to the CoAP spec (https://tools.ietf.org/html/rfc7252#section-4.5),
-			// message deduplication is relevant for CON and NON messages only
-			if ((response.getType() == Type.CON || response.getType() == Type.NON) &&
-					deduplicator.findPrevious(idByMID, exchange) != null) {
-				LOGGER.info("Duplicate response for open exchange: "+response);
-				response.setDuplicate(true);
-			} else {
-				idByMID = new KeyMID(exchange.getCurrentRequest().getMID(), null, 0);
-				exchangesByMID.remove(idByMID, exchange);
-				if (LOGGER.isLoggable(Level.FINE)) LOGGER.fine("Closed open request with "+idByMID);
+			
+			if (exchange.getCurrentRequest().isMulticast()) {
+			    // keep exchange open because we might receive additional responses
+			    LOGGER.log(Level.FINER, "Received response for multicast request: {0}", response);
+			}
+			else {
+				// we have received a Response matching the token of an ongoing Exchange's Request
+				// according to the CoAP spec (https://tools.ietf.org/html/rfc7252#section-4.5),
+				// message deduplication is relevant for CON and NON messages only
+				if ((response.getType() == Type.CON || response.getType() == Type.NON) &&
+						deduplicator.findPrevious(idByMID, exchange) != null) {
+					LOGGER.info("Duplicate response for open exchange: "+response);
```

Comment:
The `idByMID `for CON or NON are create in the remote namespace, they include the senders address.
So if this `idByMID ` is used for deduplication, this shouldn't be in conflict with the multicast.

Reply:
You are right, this part isn't in conflict with the multicast.